src/systemd-sonic-generator/systemd-sonic-generator.c

```diff
@@ -266,43 +316,176 @@ static int install_unit_file(char* unit_file, char* target, char* install_dir) {
         return -1;
     }
 
+    return 0;
+
+}
+
+
+static int install_unit_file(char* unit_file, char* target, char* install_dir) {
+    /***
+    Creates a symlink for a unit file installation
+
+    For a given unit file and target directory,
+    create the appropriate symlink in the target directory
+    to enable the unit and have it started by Systemd
+
+    If a multi ASIC platform is detected, enables multi-instance
+    services as well
+    ***/
+    char* target_instance;
+    char* prefix;
+    char* suffix;
+    int r;
+
+    assert(unit_file);
+    assert(target);
+            
+
+    if ((num_asics > 1) && strstr(unit_file, "@") != NULL) {
+
+        for (int i = 0; i < num_asics; i++) {
+
+            if (strstr(target, "@") != NULL) {
+                target_instance = insert_instance_number(target, i);
+            }
+            else {
+                target_instance = strdup(target);
+            }
+
+            r = create_symlink(unit_file, target_instance, install_dir, i);
+            if (r < 0) 
+                fprintf(stderr, "Error installing %s for target %s\n", unit_file, target_instance);
+            
+            free(target_instance);
+
+        } 
+    }
+    else {
+        r = create_symlink(unit_file, target, install_dir, -1);
+        if (r < 0) 
+            fprintf(stderr, "Error installing %s for target %s\n", unit_file, target);
+    }
+
     return 0;
 }
 
 
+static int get_num_of_asic() {
+    /***
+    Determines if the current platform is single or multi-ASIC
+    ***/
+    FILE *fp;
+    FILE *env_fp;
+    char *line = NULL;
+    char* token;
+    char* platform;
+    size_t len = 0;
+    ssize_t nread;
+    bool ans;
+    char asic_file[512];
+    char* str_num_asic;
+    int num_asic = 1;
+
+    fp = fopen(MACHINE_CONF_FILE, "r");
+
+    if (fp == NULL) {
+        fprintf(stderr, "Failed to open %s\n", MACHINE_CONF_FILE);
+        exit(EXIT_FAILURE);
+    }
+
+    while ((nread = getline(&line, &len, fp)) != -1) {
+        if (strstr(line, "onie_platform") != NULL) {
```

Comment:
we have aboot_platform and onie_platform, need to support both.

Reply:
done